python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -437,12 +469,27 @@ def current_size(self) -> int:
         return self.num_pending_actors() + self.num_running_actors()
 
     def num_running_actors(self) -> int:
-        return len(self._num_tasks_in_flight)
+        return len(self._running_actors)
+
+    def num_restarting_actors(self) -> int:
+        """Restarting actors are all the running actors not in ALIVE state."""
+        return sum(
+            running_actor_state.actor_state != gcs_pb2.ActorTableData.ActorState.ALIVE
+            for running_actor_state in self._running_actors.values()
```

Comment:
This is not a safe assumption -- if actor isn't ALIVE it doesn't necessarily mean it's restarting (it could also be DEAD for ex)

Reply:
actually the actors can be either ALIVE or RESTARTING here because they will restart indefinitely until execution finishes. 
So I'd prefer reverting `actor_state` back to `is_restarting` and add an assertion in `update_running_actor_state`